rasa/importers/autoconfig.py

```diff
@@ -0,0 +1,241 @@
+            f.seek(0)
+            for line in updated_lines:
+                f.write(line)
+    except FileNotFoundError:
+        raise ValueError(f"File '{config_file_path}' does not exist.")
+
+    if autoconfigured:
+        autoconfigured_keys = common_utils.transform_collection_to_sentence(
+            autoconfigured
+        )
+        cli_utils.print_info(
+            f"The configuration for {autoconfigured_keys} was chosen automatically. It "
+            f"was written into the config file at `{config_file_path}`."
+        )
+
+
+def _create_and_read_config_file(config_file_path: Text) -> Dict[Text, Any]:
+    import pkg_resources
+
+    cli_utils.print_warning(
+        f"Configuration file {config_file_path} does not exist or is empty or invalid. "
+        f"Creating a new one now and filling it with the current configuration."
+    )
+
+    empty_config_file = pkg_resources.resource_filename(
+        "rasa.cli.initial_project", "config.yml"
+    )
+    shutil.copy(empty_config_file, config_file_path)
+
+    return io_utils.read_config_file(config_file_path)
+
+
+def _add_missing_config_keys_to_file(
+    missing_keys: List, config_file_path: Text
+) -> None:
+    if missing_keys:
+        with open(config_file_path, "a", encoding=io_utils.DEFAULT_ENCODING) as f:
+            for key in missing_keys:
+                f.write(f"{key}:\n")
+
+
+def _get_lines_including_autoconfig(
+    lines: List[Text],
+    autoconfig_lines: Dict[Text, List[Text]],
+    language_to_overwrite: Text = None,
+) -> List[Text]:
+    autoconfigured = autoconfig_lines.keys()
+
+    lines_with_autoconfig = []
+    remove_comments_until_next_uncommented_line = False
+    for line in lines:
+        insert_section = None
+
+        # overwrite language if necessary
+        if language_to_overwrite and re.match("language:", line):
+            line = f"language: {language_to_overwrite}\n"
+
+        # remove old auto configuration
+        if remove_comments_until_next_uncommented_line:
+            if re.match("#", line):
+                continue
+            remove_comments_until_next_uncommented_line = False
+
+        # add an explanatory comment to auto configured sections
+        for key in autoconfigured:
+            if re.match(f"{key}:( *)", line):  # start of next auto-section
+                line = line + COMMENTS_FOR_KEYS[key]
+                insert_section = key
+                remove_comments_until_next_uncommented_line = True
+
+        lines_with_autoconfig.append(line)
+
+        if not insert_section:
+            continue
+
+        # add the auto configuration (commented out)
+        lines_with_autoconfig += autoconfig_lines[insert_section]
+
+    return lines_with_autoconfig
+
+
+def _get_commented_out_autoconfig_lines(
+    config: Dict[Text, Any], autoconfigured: List[Text]
+) -> Dict[Text, List[Text]]:
+    import ruamel.yaml as yaml
+
+    yaml_parser = yaml.YAML()
+    yaml_parser.indent(mapping=2, sequence=4, offset=2)
```

Comment:
what about using `yaml.dump`? We can split everything by `/n`, prefix with `# ` and we got the same result? Or use https://yaml.readthedocs.io/en/latest/example.html#output-of-dump-as-a-string . This way we don't need a file